common/src/main/java/com/microsoft/identity/common/internal/authorities/AzureActiveDirectoryAudience.java

```diff
@@ -55,6 +67,63 @@ public String getTenantId() {
         return mTenantId;
     }
 
+
+    /**
+     *
+     * Must me called on a worker thread.
+     *
+     * Method which queries the {@link OpenIdProviderConfiguration}
+     * to get tenant UUID for the authority with tenant alias
+     *
+     * @return : tenant UUID
+     * @throws ServiceException
+     * @throws ClientException
+     */
+    @WorkerThread
+    public String getTenantUuidForAlias()
+            throws ServiceException, ClientException {
+        // if the tenant id is already a UUID, return
+        if(StringUtil.isUuid(mTenantId)){
+            return mTenantId;
+        }
+
+        final OpenIdProviderConfiguration providerConfiguration =
+                loadOpenIdProviderConfigurationMetadata(mCloudUrl);
+
+        final String issuer = providerConfiguration.getIssuer();
+        final Uri issuerUri = Uri.parse(issuer);
+        final List<String> paths = issuerUri.getPathSegments();
+
+        if (paths.isEmpty()) {
+            final String errMsg = "OpenId Metadata did not contain a path to the tenant";
+
+            com.microsoft.identity.common.internal.logging.Logger.error(
+                    TAG,
+                    errMsg,
+                    null
+            );
+
+            throw new ClientException(errMsg);
+        }
+
+        return paths.get(0);
```

Comment:
nit: to make this future-proof, should we also invoke StringUtil.isUuid again on this value?